hermes/probe/create/create_file_test.go

```diff
@@ -0,0 +1,140 @@
+			}
+			if err.Error() != test.wantErr {
+				t.Errorf("{%v, %v}.FileName() gave wrong error expected %s, got %s", test.file.ID, test.file.Size, test.wantErr, err.Error())
+			}
+		}
+	}
+}
+
+func TestChecksum(t *testing.T) {
+	f := RandomHermesFile{11, 100}
+	fTwo := RandomHermesFile{13, 1000}
+	fCopy := RandomHermesFile{11, 100}
+	checksum, err := f.CheckSum()
+	if err != nil {
+		t.Error(err)
+	}
+	checksumTwo, err := fTwo.CheckSum()
+	if err != nil {
+		t.Error(err)
+	}
+
+	checksumCopy, err := fCopy.CheckSum()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if fmt.Sprintf("%x", checksum) != fmt.Sprintf("%x", checksumCopy) { //comparing checksums converted to strings as a slice can only be compared to nil
+		t.Errorf("Checksum returned different values for the same RandomHermesFiles {%v, %v}", f.ID, f.Size)
+	}
+
+	if fmt.Sprintf("%x", checksum) == fmt.Sprintf("%x", checksumTwo) {
+		t.Errorf("Checksum returned the same value for two different RandomHermesFiles {%v, %v} and {%v, %v}", f.ID, f.Size, fTwo.ID, fTwo.Size)
+	}
+
+}
+
+func TestCreateFile(t *testing.T) {
+	ctx := context.Background()
+	bucketName := "test_bucket_probe0"
+	client := mock.NewFakeClient()
+	fbh := client.Bucket(bucketName)                         // fakeBucketHandle
+	if err := fbh.Create(ctx, bucketName, nil); err != nil { // creates the bucket with name "test_bucket_probe0"
+		t.Error(err)
+	}
+	fileID := int32(6)
+	fileSize := 50
+	target := &probe.Target{
+		&probepb.Target{
+			Name:                   "hermes",
+			TargetSystem:           probepb.Target_GOOGLE_CLOUD_STORAGE,
+			TotalSpaceAllocatedMib: int64(1000),
+			BucketName:             "test_bucket_probe0",
+		},
+		&journalpb.StateJournal{
+			Filenames: make(map[int32]string),
+		},
+		&metrics.Metrics{},
+	}
+	hp := &probepb.HermesProbeDef{
+		ProbeName: proto.String("createfile_test"),
+		Targets: []*probepb.Target{
+			&probepb.Target{
+				Name:                   "hermes",
+				TargetSystem:           probepb.Target_GOOGLE_CLOUD_STORAGE,
+				TotalSpaceAllocatedMib: int64(100),
+				BucketName:             "test_bucket_probe0",
+			},
+		},
+		TargetSystem: probepb.HermesProbeDef_GCS.Enum(),
+		IntervalSec:  proto.Int32(3600),
+		TimeoutSec:   proto.Int32(60),
+		ProbeLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.1,0.2,0.4,0.6,0.8,1.6,3.2,6.4,12.8,1",
+			},
+		},
+		ApiCallLatencyDistribution: &metricpb.Dist{
+			Buckets: &metricpb.Dist_ExplicitBuckets{
+				ExplicitBuckets: "0.1,0.2,0.4,0.6,0.8,1.6,3.2,6.4,12.8,1",
+			},
+		},
+	}
+	probeTarget := &probepb.Target{
+		Name:                   "hermes",
+		TargetSystem:           probepb.Target_GOOGLE_CLOUD_STORAGE,
+		TotalSpaceAllocatedMib: int64(100),
+		BucketName:             "test_bucket_probe0",
+	}
+
+	err := fmt.Errorf("") //err cannot be defined in the if statement below using ':=' as then the set up fails with the error:  expected identifier on left side of :=
```

Comment:
You want `var err error` instead. Also, remove the comment.

Reply:
changed and removed!